mappings/platform_file_centric.json

```diff
@@ -0,0 +1,317 @@
+                "type": "keyword"
+              }
+            }
+          },
+          "experiment": {
+            "type": "object",
+            "properties": {
+              "platform": {
+                "type": "keyword"
+              },
+              "library_strategy": {
+                "type": "keyword"
+              }
+            }
+          }
+        }
+      },
+      "file": {
+        "properties": {
+          "id": {
+            "type": "keyword"
+          },
+          "size": {
+            "type": "integer"
+          },
+          "md5sum": {
+            "type": "keyword"
+          },
+          "name": {
+            "type": "keyword",
+            "copy_to": ["file_autocomplete"]
+          }
+        }
+      },
+      "index_file": {
+        "properties": {
+          "object_id": {
+            "type": "keyword"
+          },
+          "format": {
+            "type": "keyword"
+          },
+          "md5sum": {
+            "type": "keyword"
+          },
+          "name": {
+            "type": "keyword"
+          },
+          "size": {
+            "type": "integer"
+          }
+        }
+      },
+      "donors": {
+        "type": "nested",
+        "properties": {
+          "id": {
+            "type": "keyword",
+            "copy_to": ["file_autocomplete"]
+          },
+          "submitter_donor_id": {
+            "type": "keyword",
+            "copy_to": ["file_autocomplete"]
+          },
+          "gender": {
+            "type": "keyword"
+          },
+          "vital_status": {
+            "type": "keyword"
+          },
+          "survival_time": {
+            "type": "integer"
+          },
+          "prior_malignancy": {
+            "type": "keyword"
+          },
+          "primary_diagnosis": {
+            "properties": {
+              "id": {
+                "type": "keyword"
+              },
+              "age_at_diagnosis": {
+                "type": "integer"
+              },
+              "cancer_type_code": {
+                "type": "keyword"
+              },
+              "clinical_tumour_staging_system": {
+                "type": "keyword"
+              },
+              "clinical_t_category": {
+                "type": "keyword"
+              },
+              "clinical_n_category": {
+                "type": "keyword"
+              },
+              "clinical_m_category": {
+                "type": "keyword"
+              }
+            }
+          },
+          "follow_ups": {
+            "type": "nested",
+            "properties": {
+              "id": {
+                "type": "keyword"
+              },
+              "disease_status_at_followup": {
+                "type": "keyword"
+              },
+              "relapse_type": {
+                "type": "keyword"
+              }
+            }
+          },
+          "treatments": {
+            "type": "nested",
+            "properties": {
+              "id": {
```

Comment:
this is a submitter id by the way, same for follow up

Reply:
do you mean we `also` need to add submitter id?  i think we need to keep the db id in here for uniquess constraint no?

Reply:
there is no db id per treatment. I think this should be the submitter Id.. and no elastic only requires a unique id per document (file id in this case), nested objects don't need unique id.

Reply:
so i will just point out that in the mockup we had `pretty argo ids`  for these objects too: 
![image](https://user-images.githubusercontent.com/15824252/78581577-0ce7b180-7802-11ea-9afc-39bd71e6bf09.png)
 
I understand now this is not the actual case, but we would like them. how hard would this be to do?

Reply:
I think it's a 5 points ticket

Reply:
@christinayung thoughts from you on this ^ ?

Reply:
We do ask for `submitter_treatment_id` and `submitter_follow_up_id`, but there's no guarantee that the uniqueness that we need will be there.  So best to assign our own IDs

Reply:
there is no guarantee across different programs, but it's unique in the same program because our db refuses duplication there. however we did it for specimens and samples, so I don't see why these can't have it too

Reply:
on a side note, the UI is a bit weird showing treatments not therapies though

Reply:
and our index is only indexing treatments not therapies

Reply:
@blabadi therapies are only relatd to treatments.  they show within the treatment details when you click on a treatment.

Reply:
Also, this is the index for the file repository - i believe we discussed a Get Donor API from clinical for the gateway with the donor page, not this index?  therapies do not need to be available from the file repository.

Reply:
@blabadi I made a ticket for the id implementation: https://github.com/icgc-argo/argo-clinical/issues/460 

Are you ok to leave both `id` and `submitter_<enitty>_id` in the index for now and when we first index if they need to be blank leaving them blank?

Reply:
yeah sure